### PR TITLE
Add POC telem lib integration

### PR DESCRIPTION
**Pull request by boundsj**
This PR illustrates how the new telem lib can be used in this project.

cc @ericrwolfe @bsudekum

### CODE REVIEW

File: MapboxCoreNavigation/RouteController.swift [hunk 1]
```diff
@@ -139,6 +142,11 @@ open class RouteController: NSObject {
         super.init()
         
         self.locationManager.delegate = self
+        
+        // This is just a test!!! (needs a real access token, a user agent base, and the actual nav SDK version
+        events.initialize(withAccessToken: "your token here!", userAgentBase: "MapboxEventsiOSNavigation", hostSDKVersion: "42.42.42")
+        events.isDebugLoggingEnabled = true
+        events.sendTurnstileEvent()
```
Comment (ericrwolfe) [inline MapboxCoreNavigation/RouteController.swift]: This seems to send a generic `MMEEventTypeAppUserTurnstile`. Do we want these turnstile events to be specific to maps, navigation, etc?

Comment (bsudekum) [inline MapboxCoreNavigation/RouteController.swift]: I think @boundsj might just be showing an example of how you would use the API here?

Comment (boundsj) [inline MapboxCoreNavigation/RouteController.swift]: @bsudekum is correct, this was just an example. However, as discussed, we are implementing a universal turnstile event that all host SDKs will use.

**Comment by bsudekum** [comment]
@boundsj we'll also need cocoapods support for this too. Can we simply checkin a podspec into https://github.com/mapbox/mapbox-telemetry-ios?

**Comment by bsudekum** [comment]
Closing in favor of https://github.com/mapbox/mapbox-navigation-ios/pull/304

**Pull request closed without merging**